app/services/valid_test_data_generators/applications_populater.rb

```diff
@@ -94,17 +110,48 @@ def reject_application(application)
       application.reload
     end
 
+    def defer_application(application)
+      Participants::Defer.new(lead_provider:,
+                              participant: application.user,
+                              course_identifier: application.course.identifier,
+                              reason: Participants::Defer::DEFERRAL_REASONS.sample).defer
+    end
+
+    def withdrawn_application(application)
+      Participants::Withdraw.new(lead_provider:,
+                                 participant: application.user,
+                                 course_identifier: application.course.identifier,
+                                 reason: Participants::Withdraw::WITHDRAWAL_REASONS.sample).withdraw
+    end
+
     def create_declarations(application)
       %w[started retained-1 retained-2 completed].each do |declaration_type|
         break if Faker::Boolean.boolean(true_ratio: 0.5)
 
         FactoryBot.create(
           :declaration,
-          :submitted_or_eligible,
+          state: Declaration.states.keys.sample,
           application:,
           declaration_type:,
         )
       end
     end
+
+    def create_outcomes(application)
+      completed_declaration = application.declarations.eligible_for_outcomes(lead_provider, application.course.identifier).first
+      return unless completed_declaration
+      return unless CourseGroup.joins(:courses).leadership_or_specialist.where(courses: { identifier: application.course.identifier }).exists?
+
+      %i[passed voided failed].each do |state_trait|
```

Comment:
could we use the create states similar to https://github.com/DFE-Digital/npq-registration/pull/1519/files#diff-38f0cbe9d749dd85fe13a256c7c104dfc5a070dbf1a7e0e205b6c9b80df3ff19R115 :)